app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -64,27 +77,80 @@ internal class DefaultMetricsStorage(
             Event.GrowthData.FirstUriLoadForDay -> {
                 settings.uriLoadGrowthLastSent = System.currentTimeMillis()
             }
+            Event.GrowthData.FirstWeekSeriesActivity -> {
+                settings.firstWeekSeriesGrowthSent = true
+            }
         }
     }
 
+    private fun updateDaysOfUse() {
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData
+        val currentDate = Calendar.getInstance()
+        val asString = dateFormatter.format(currentDate.time)
+        if (currentDate.timeInMillis.withinFirstWeek() && daysOfUse.none { it == asString }) {
+            settings.firstWeekDaysOfUseGrowthData = daysOfUse.plus(asString)
+        }
+    }
+
+    private fun shouldTrackFirstWeekActivity(): Boolean = Result.runCatching {
+        if (!System.currentTimeMillis().withinFirstWeek() || settings.firstWeekSeriesGrowthSent) {
+            return false
+        }
+
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData.map {
+            dateFormatter.parse(it)
+        }.sorted()
+
+        // This loop will check whether the existing list of days of use, combined with the
+        // current date, contains any periods of 3 days of use in a row. It's mostly this ugly
+        // because of the inherent mutability of the Calendar type, but hopefully we can rip
+        // this code out soon™ anyway
+        for (idx in daysOfUse.indices) {
+            if (idx + 1 > daysOfUse.lastIndex || idx + 2 > daysOfUse.lastIndex) {
+                continue
+            }
+            val day1 = daysOfUse[idx]!!.time.toCalendar()
+            val day2 = daysOfUse[idx + 1]!!.time.toCalendar()
+            val day3 = daysOfUse[idx + 2]!!.time.toCalendar()
+            var areSequential = true
+            day1.add(Calendar.DAY_OF_MONTH, 1)
+            areSequential = areSequential && day1 == day2
+            day1.add(Calendar.DAY_OF_MONTH, 1)
+            areSequential = areSequential && day1 == day3
```

Comment:
```suggestion
            val referenceDate = daysOfUse[idx]!!.time.toCalendar()
            val oneDayAfterReference = referenceDate.also { it.add(Calendar.DAY_OF_MONTH, 1) }
            val twoDaysAfterReference = referenceDate.also { it.add(Calendar.DAY_OF_MONTH, 2) }
            val secondDateEntry = daysOfUse[idx + 1]!!.time.toCalendar()
            val thirdDateEntry = daysOfUse[idx + 2]!!.time.toCalendar()
            var areSequential = true
            areSequential = areSequential && oneDayAfterReference == secondDateEntry
            areSequential = areSequential && twoDaysAfterReference == thirdDateEntry
```

Thoughts on rewording some things to make it a bit clearer what's being compared?

Reply:
I do like the name changes, I'll work on updating to something similar. The real issue here is that `Calendar::add` mutates the underlying object instead of returning a new one with the updated date, so things like `referenceDate.also { it.add(Calendar.DAY_OF_MONTH, 1) }` don't actually work. 

I worked around that with a `copy` extension in the test file but thought it might be bloat-y in the actual code. Do you think it's worth including here as well?

Reply:
I've included an extension function to clean up the creation of a new Calendar object for the subsequent day

Reply:
This looks much better!
![image](https://user-images.githubusercontent.com/87384386/203614387-a7a8e0a6-ceb9-413f-b4a4-18e6bdc43390.png)